bin/ream/src/main.rs

```diff
@@ -49,19 +58,26 @@ async fn main() {
 
             info!("ream database initialized ");
 
-            match Network::init(async_executor, &binding).await {
-                Ok(mut network) => {
-                    main_executor.spawn(async move {
-                        network.polling_events().await;
-                    });
+            let network_future = async {
+                match Network::init(async_executor, &binding).await {
+                    Ok(mut network) => {
+                        main_executor.spawn(async move {
+                            network.polling_events().await;
+                        });
 
-                    tokio::signal::ctrl_c().await.unwrap();
-                }
-                Err(e) => {
-                    error!("Failed to initialize network: {}", e);
-                    return;
+                        tokio::signal::ctrl_c().await.unwrap();
+                    }
+                    Err(e) => {
+                        error!("Failed to initialize network: {}", e);
+                    }
                 }
+            };
+
+            tokio::select! {
+                _ = http_future => {},
+                _ = network_future => {},
             }
+            tokio::signal::ctrl_c().await.unwrap();
```

Comment:
```suggestion
            tokio::signal::ctrl_c().await.unwrap();
```

could we use expect instead of unwrap here